### PR TITLE
blockproofkey-registration

### PR DESCRIPTION
**Pull request by algonathan**
## Summary
key registration transaction includes the blockProofKey

### CODE REVIEW

File: data/basics/userBalance.go [hunk 1]
```diff
@@ -146,8 +146,9 @@ type AccountData struct {
 	// the past week".
 	RewardedMicroAlgos MicroAlgos `codec:"ern"`
 
-	VoteID      crypto.OneTimeSignatureVerifier `codec:"vote"`
-	SelectionID crypto.VRFVerifier              `codec:"sel"`
+	VoteID       crypto.OneTimeSignatureVerifier `codec:"vote"`
+	SelectionID  crypto.VRFVerifier              `codec:"sel"`
+	BlockProofID crypto.VerifyingKey             `codec:"blockproof"`
```
Comment (tsachiherman) [inline data/basics/userBalance.go:151]: consider using a shorter codec field name. This field name is included in every single msgp encoding of the account data, which is quite frequent. maybe `blkprf` ?

File: data/transactions/transaction.go [hunk 2]
```diff
@@ -291,6 +293,15 @@ func (tx Transaction) WellFormed(spec SpecialAddresses, proto config.ConsensusPa
 			}
 		}
 
+		if proto.EnableBlockProofKeyregCheck {
```
Comment (tsachiherman) [inline data/transactions/transaction.go:296]: I think that you want to ensure that this `BlockProofPK` is set on online key registration transaction, and clear on offline key registration transaction. ( but I could be wrong here ).

File: data/basics/userBalance.go (hunk 1 above)
Comment (algonathan) [reply data/basics/userBalance.go]: Thanks for the comments! 
i've changed it

**Review by algoidan — COMMENTED** [review]
please consider those remarks

**Review by algoidan — APPROVED** [review]
great job!

**Review by algoidan — COMMENTED** [review]
V

**Pull request merged**